tv_apps/browser/js/browser_dialog.js

```diff
@@ -0,0 +1,554 @@
+ * The dialog listen to mozbrowsershowmodalprompt event.
+ * (alert/confirm/prompt)
+ *
+ * @namespace Dialog
+ */
+var BrowserDialog = {
+  // Dialog Event
+  dialogEvt: null,
+  browserDialogInputArea: null,
+  PORT_MIN: 0,
+  PORT_MAX: 65535,
+  focusElement: [],
+  focusIndex: {x:0, y:0},
+  defaultFocusIndex: {x:0, y:0},
+
+  /** Get all elements when inited. */
+  getAllElements: function dialog_getAllElements() {
+    var toCamelCase = function toCamelCase(str) {
+      return str.replace(/\-(.)/g, function replacer(str, p1) {
+        return p1.toUpperCase();
+      });
+    };
+
+    var elementIDs = [
+      'browser-dialog-base', 'browser-dialog',
+      'browser-dialog-title', 'browser-dialog-msg',
+      'browser-dialog-input',
+      'browser-dialog-input-area',
+      'browser-dialog-input-clear',
+      'browser-dialog-button', 'browser-dialog-button-margin',
+      'browser-dialog-button1', 'browser-dialog-button2',
+    ];
+
+    // Loop and add element with camel style name to Dialog attribute.
+    elementIDs.forEach(function createElementRef(name) {
+      this[toCamelCase(name)] = document.getElementById(name);
+    }, this);
+  },
+
+  /** Initialization. Get DOM elements and add listeners. */
+  init: function dialog_init() {
+    // Get all elements initially.
+    this.getAllElements();
+
+    this.browserDialogBase.addEventListener('mouseup',
+        this.dialogBaseClick.bind(this));
+
+    // Add event listener(dialog)
+    this.browserDialogButton1.addEventListener('mouseup',
+        this.dialogButton1.bind(this));
+    this.browserDialogButton1.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+
+    this.browserDialogButton2.addEventListener('mouseup',
+        this.dialogButton2.bind(this));
+    this.browserDialogButton2.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+
+    this.browserDialogInput.addEventListener('submit',
+        this.dialogInput.bind(this));
+    this.browserDialogInputClear.addEventListener('mouseup',
+        this.dialogInputClear.bind(this));
+    this.browserDialogInputClear.addEventListener('keyup',
+        this.dialogButtonKeyup.bind(this));
+
+    this.browserDialogInputArea.addEventListener('blur',
+        this.dialogInputAreaBlur.bind(this));
+    this.browserDialogInputArea.addEventListener('focus',
+        this.dialogInputAreaFocus.bind(this));
+    this.browserDialogInputArea.addEventListener('mousedown',
+        this.dialogInputAreaClick.bind(this));
+  },
+
+  /**
+   * create Dialog
+   */
+  createDialog: function dialog_createDialog(type, evt) {
+    var opt;
+
+    this.cancelDialog();
+    if( evt != null ) {
+      evt.preventDefault();
+    }
+    this.dialogEvt = evt;
+    this.browserDialogBase.style.display = 'block';
+
+    switch(type) {
+      case 'del_cookie':
+        opt = {
+          title: null,
+          msg: _('WB_LT_CONFIRM_DELETE_COOKIES'),
+          bt1: _('LT_CANCEL'),
+          bt2: _('WB_LT_CLEAR')
```

Comment:
this whole createDialog function should create opt with bt1, bt2 being l10nIDs, and then assign them to data-l10n-id in UI.